### PR TITLE
MGMT-11635:Bad request error when user chooses both LVM and CNV

### PR DESCRIPTION
**Pull request by Mrudraia**
## List all the issues related to this PR

- [ ] New Feature 
- [X] Enhancement 
- [ ] Bug fix
- [ ] Tests
- [ ] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [ ] Automation (CI, tools, etc)
- [ ] Cloud
- [ ] Operator Managed Deployments
- [x] None

## How was this code tested?

- [ ] assisted-test-infra environment
- [X] dev-scripts environment
- [ ] Reviewer's test appreciated
- [ ] Waiting for CI to do a full test run
- [ ] Manual (Elaborate on how it was tested)
- [ ] No tests needed

## Assignees

/cc @avishayt
/cc @rewantsoni  

## Checklist

- [ ] Title and description added to both, commit and PR.
- [ ] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [ ] Reviewers have been listed
- [ ] This change does not require a documentation update (docstring, `docs`, README, etc)
- [ ] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- Are the title and description (in both PR and commit) meaningful and clear?
- Is there a bug required (and linked) for this change?
- Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md

### CODE REVIEW

File: internal/bminventory/inventory.go [hunk 1]
```diff
@@ -2593,6 +2593,10 @@ func (b *bareMetalInventory) updateOperatorsData(ctx context.Context, cluster *c
 	}
 
 	for _, updatedOperator := range updateOLMOperators {
+		if updatedOperator.Name == "LVM" || updatedOperator.Name == "CNV" {
+			err = errors.Wrapf(err, "failed, either LVM or CNV operator can be installed, both can not be installed")
```
Comment (gamli75) [inline internal/bminventory/inventory.go:2597]:
```suggestion
			err = errors.Wrapf(err, "Currently, you can not install OpenShift Data Foundation Logical Volume Manager operator at the same time as Virtualization operator.")
```

Comment (gamli75) [reply internal/bminventory/inventory.go:2597]: Please use the same text as we will display in the UI, see: https://issues.redhat.com/browse/MGMT-11145

Comment (avishayt) [reply internal/bminventory/inventory.go:2597]: This:
Installing OpenShift Data Foundation Logical Volume Manager together with OpenShift Virtualization is not currently allowed.

**Comment by avishayt** [comment]
1. The code doesn't take care of all cases - need to ensure that these two operators are never enabled together (e.g., both enabled upon cluster register, one enabled and then the other enabled upon cluster update, both enabled in cluster update).
2. Please add unit tests.

**Review by avishayt — CHANGES_REQUESTED** [review]
The current change works fine when updating a cluster, but doesn't cover RegisterCluster.  We discussed moving this code to a help function and calling it from both places.

Also, please add unit tests.

File: internal/bminventory/inventory.go [hunk 2]
```diff
@@ -2583,6 +2589,7 @@ func (b *bareMetalInventory) updateClusterCPUFeatureUsage(cluster *common.Cluste
 }
 
 func (b *bareMetalInventory) updateOperatorsData(ctx context.Context, cluster *common.Cluster, params installer.V2UpdateClusterParams, usages map[string]models.Usage, db *gorm.DB, log logrus.FieldLogger) error {
+	// var enabledoperators []*models.MonitoredOperator
```
Comment (rewantsoni) [inline internal/bminventory/inventory.go:2592]: We can remove the comment that is not required.

File: internal/bminventory/inventory.go [hunk 3]
```diff
@@ -472,8 +472,14 @@ func (b *bareMetalInventory) RegisterClusterInternal(
 		if err != nil {
 			return nil, err
 		}
+		err = operators.EnsureLVMAndCNVNotEnabled(newOLMOperators)
+		if err != nil {
+			log.Error(err)
+			return nil, common.NewApiError(http.StatusInternalServerError, err)
```
Comment (avishayt) [inline internal/bminventory/inventory.go:478]:
```suggestion
			return nil, common.NewApiError(http.StatusBadRequest, err)
```

File: internal/operators/manager_test.go [hunk 4]
```diff
@@ -161,6 +178,36 @@ var _ = Describe("Operators manager", func() {
 		)
 	})
 
+	Context("EnsureLVMAndCNVNotEnabled", func() {
```
Comment (avishayt) [inline internal/operators/manager_test.go:181]: What do these unit tests check?  They should be calling RegisterCluster and UpdateCluster, and making sure that the proper errors are returned.

**Pull request closed without merging**